src/tracking.c

```diff
@@ -388,7 +388,7 @@ void trackingInvalidateKey(client *c, robj *keyobj, int bcast) {
         /* If the client enabled the NOLOOP mode, don't send notifications
          * about keys changed by the client itself. */
         if (target->flags & CLIENT_TRACKING_NOLOOP &&
-            target == c)
+            target == server.current_client)
```

Comment:
in other places (tracking included), we do it like this:
```
        client *caller = (c->flags & CLIENT_LUA && server.lua_caller) ?
                            server.lua_caller : c;
```
i wonder if there are any differences?
maybe with modules doing RM_Call?
i.e. matching it to current_client will mean that we will still look at the client that initiated the module command, and matching it to my suggestion would mean we look at the module fake client.

@MeirShpilraien do you heave any input here?

Reply:
I believe `current_client` is better than `lua_caller`, Lua caller could be a module client that called the Lua script.

Notice that this will not work if called using RM_Call from a background thread (`server.current_client`  will be NULL). But currently its not working even if RM_Call is done on the main thread, so I believe this is better than what we have today.